netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -283,6 +283,15 @@ public void operationComplete(ChannelFuture future) throws Exception {
         eventLoopReferenceCounter.release();
       }
     });
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
Why this condition? Is it an optimization? Honestly, I'd prefer not to optimize this case, which also has benefits like the FINE logging happening independent of how interrupt races.

Reply:
IIRC normal futures throw if they try to block while the thread is interrupted, but I wasn't sure Netty's would do the same.  I'm happy to remove it